waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/StaticDatabaseMappingService.java

```diff
@@ -155,6 +169,21 @@ private void validateFederatedMetastoreDatabases(List<String> mappableDatabases,
     }
   }
 
+  private List<String> getDatabasesFromPattern(List<String> allDatabases, List<String> mappedDatabases) {
+    List<String> matchedDatabases = new ArrayList<String>();
+    for (String mappedDatabase : mappedDatabases) {
+      try {
+        Iterable<String> matches = Iterables.filter(allDatabases, Predicates.contains(Pattern.compile(mappedDatabase)));
```

Comment:
You're right, using `Whitelist` makes it better. 
I thought that the only thing that does stuff with regex was `GrammarUtils` and didn't see how it could be used in `getDatabasesFromPattern`.